### PR TITLE
feature: sorting only added

### LINKED ISSUE #115: Is only sorting possible?
**Is your feature request related to a problem? Please describe.**
It's not a problem, rather a request. Would it be possible to just sort instead of converting? For example, it would be really cool to have the possibility to sort by height or width of an image.

**Describe the solution you'd like**
Add a sorting function.

**Describe alternatives you've considered**
None.

### PR DESCRIPTION
**Pull request by Probotus**
changes:
- [feat: various providers added](https://github.com/BoBoBaSs84/DDS.Tools/commit/e449ba5420149d9923e283b425833127c1701032)
- [feat: directory provider improved](https://github.com/BoBoBaSs84/DDS.Tools/commit/3019976e125e82eed555545231afe04b31170658)
- [feat: providers even more improved](https://github.com/BoBoBaSs84/DDS.Tools/commit/642f4c51fb0f1d8e3271ad400b5bacfa4454343c)
- [feat: commands improved](https://github.com/BoBoBaSs84/DDS.Tools/commit/4278a8fa49229db2784f2940e4a313bb11a7586a)
- [feat: enumerator added and todo service improved](https://github.com/BoBoBaSs84/DDS.Tools/commit/b30636246e132381e6c76c016f7f9c77a3f82229)
- [feat: unit tests added and fixed](https://github.com/BoBoBaSs84/DDS.Tools/commit/6f34d0149f35c787d7e5b420e9695b9c20c413fa)
- [fix: review changes applied](https://github.com/BoBoBaSs84/DDS.Tools/pull/116/commits/a9e148a627f5a62fca6d8d841de8798787d9f465)

closes #115

### CODE REVIEW

**Comment by BoBoBaSs84** [comment]
I have to Test this in depth manually ..

File: src/DDS.Tools/Enumerators/ConvertModeType.cs [hunk 1]
```diff
@@ -13,5 +13,10 @@ public enum ConvertModeType : byte
 	/// <summary>
 	/// The manual mode.
 	/// </summary>
-	Manual = 1
+	Manual = 1,
+
+	/// <summary>
+	/// The sorting mode.
+	/// </summary>
+	Sorting = 2
```
Comment (BoBoBaSs84) [inline src/DDS.Tools/Enumerators/ConvertModeType.cs:21]: Grouping?

File: src/DDS.Tools/Services/TodoService.cs [hunk 2]
```diff
@@ -150,41 +155,79 @@ private void GetTodoDone(TodoModel todo, ConvertSettingsBase settings, ImageType
...
+			_todosDone.Add(todo.FileHash);
+			return;
+		}
 
 		SaveImage(settings, todo, imageType);
 
 		_todosDone.Add(todo.FileHash);
 	}
 
+	private void MoveImage(ConvertSettingsBase settings, TodoModel todo, ImageType imageType)
```
Comment (BoBoBaSs84) [inline src/DDS.Tools/Services/TodoService.cs:177]: Copy instead of move?

File: tests/DDS.ToolsTests/Services/TodoServiceTests.GetTodosDone.cs [hunk 3]
```diff
@@ -43,4 +43,20 @@ public void GetTodosDoneManualTest()
 
 		s_todoService.GetTodosDone(todos, settings, imageType);
 	}
+
+	[TestMethod]
+	public void GetTodosDoneSortingTest()
```
Comment (BoBoBaSs84) [inline tests/DDS.ToolsTests/Services/TodoServiceTests.GetTodosDone.cs:48]: Grouping?

**Review by BoBoBaSs84 — CHANGES_REQUESTED** [review]
So far so good, it really should be named grouping.
The docs are missing, I'll open up a new issue for that.

**Comment by Probotus** [comment]
I applied the changes.

**Review by BoBoBaSs84 — APPROVED** [review]

**Pull request merged**